ncm-spma/src/main/perl/spma/ips.pm

```diff
@@ -411,27 +443,42 @@ sub run_pkg_command
 
 #
 # Get pkg@ver keys from hash where ver is actually the value.
+# Can be given an existing array to amend.
 #
 sub pkg_keys
 {
-    my ($self, $hash) = @_;
+    my ($self, $hash, $keylist) = @_;
+
+    $keylist ||= [];
 
-    my @keys;
     for my $pkg (keys %$hash) {
         my $ver = "";
         $ver = "\@$hash->{$pkg}" if $hash->{$pkg} ne "";
-        push @keys, "$pkg$ver";
+        push @$keylist, "$pkg$ver";
     }
-    return \@keys;
+    return $keylist;
+}
+
+#
+# Returns >0 if pkg command on this system has exact-install capability
+# introduced in Solaris 11.2
+#
+sub pkg_has_exact_install
+{
+    my $stderr;
+    my $proc = CAF::Process->new(PKG_HELP, stderr => \$stderr);
+    $proc->execute();
+    return scalar(grep /exact-install/, split('\n', $stderr));
 }
 
 #
 # Updates IPS packages on the system.
 #
 sub update_ips
 {
-    my ($self, $pkgs, $uninst, $run_now, $allow_user_pkgs, $cmdfile, $flagfile,
-            $bename, $include_idrs, $freeze, $imagedir) = @_;
+    my ($self, $pkgs, $reject, $uninst, $whitelist,
```

Comment:
did you insert new options in the middle? not very backwards compatible (but this can be considered internal API)